service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -131,4 +140,90 @@ public DatasetId createDataset(
     ensureActionPermission(user, dataset, SamAction.CREATE_METADATA);
     return datasetDao.create(dataset).id();
   }
+
+  public DatasetPreviewMetadataResponse getDatasetPreviewMetadata(
+      AuthenticatedUserRequest user, DatasetId datasetId) {
+    var dataset = datasetDao.retrieve(datasetId);
+    ensureActionPermission(user, dataset, SamAction.READ_ANY_METADATA);
+    // Right now, this code makes an assumption that all datasets are snapshots.
+    // Eventually this will not be true and this should change.
+    var snapshotMetadata = datarepoService.getPreviewMetadata(user, dataset.storageSourceId());
+    return new DatasetPreviewMetadataResponse()
+        .id(snapshotMetadata.getId())
+        .dataProject(snapshotMetadata.getDataProject())
+        .tables(convertDatarepoTablesToCatalogTables(snapshotMetadata.getTables()));
+  }
+
+  private List<TableModel> convertDatarepoTablesToCatalogTables(
+      List<bio.terra.datarepo.model.TableModel> datarepoTables) {
+    return Optional.ofNullable(datarepoTables)
+        .map(
+            tableModels ->
+                tableModels.stream()
+                    .map(
+                        tableModel ->
+                            new TableModel()
+                                .name(tableModel.getName())
+                                .columns(
+                                    convertDataRepoColumnsToCatalogColumns(tableModel.getColumns()))
+                                .primaryKey(tableModel.getPrimaryKey())
+                                .partitionMode(
+                                    convertDatarepoPartiotionMode(tableModel.getPartitionMode()))
+                                .datePartitionOptions(
+                                    convertDatarepoDatePartionsOptionsModel(
+                                        tableModel.getDatePartitionOptions()))
+                                .intPartitionOptions(
+                                    convertDatarepoIntPartionsOptionsModel(
+                                        tableModel.getIntPartitionOptions()))
+                                .rowCount(tableModel.getRowCount()))
+                    .collect(Collectors.toList()))
```

Comment:
In JDK 17 you can use `toList()`
```suggestion
                    .toList())
```